custom_components/google_home/api.py

```diff
@@ -227,3 +229,89 @@ async def update_google_devices_information(self) -> List[GoogleHomeDevice]:
             ]
         )
         return coordinator_data
+
+    async def delete_timer_or_alarm(
+        self, device: GoogleHomeDevice, item_to_delete: str
+    ) -> None:
+        """Deletes a timer or alarm,
+        can also delete multiple if a list is provided (Not implemented yet)."""
+
+        url = self.create_url(str(device.ip_address), PORT, API_ENDPOINT_DELETE)
+
+        # We need to remove charset=UTF-8 or else it will return a 400 BAD request.
+        # I think this is because of the character "/" in the id string.
+        HEADERS.update(
+            {
+                HEADER_CAST_LOCAL_AUTH: str(device.auth_token),
```

Comment:
```suggestion
                HEADER_CAST_LOCAL_AUTH: device.auth_token,
```

Reply:
Same as the other problem. If i don't explicitly convert it to string i get this error in mypy:

```text
custom_components/google_home/api.py:245: error: Dict entry 0 has incompatible type "str": "Optional[str]"; expected "str": "str"
Found 1 error in 1 file (checked 4 source files)
custom_components/google_home/api.py:245: error: Dict entry 0 has incompatible type "str": "Optional[str]"; expected "str": "str"
Found 1 error in 1 file (checked 4 source files)
custom_components/google_home/api.py:245: error: Dict entry 0 has incompatible type "str": "Optional[str]"; expected "str": "str"
Found 1 error in 1 file (checked 1 source file)
```

Reply:
Mypy prevents you from adding a bug 🙂